src/source/Ice/IceAgent.c

```diff
@@ -1774,6 +1774,59 @@ STATUS iceAgentCheckConnectionStateSetup(PIceAgent pIceAgent)
     return retStatus;
 }
 
+STATUS updateCandidateStats(PIceAgent pIceAgent, BOOL isRemote)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    CHK(pIceAgent != NULL && pIceAgent->pDataSendingIceCandidatePair != NULL, STATUS_NULL_ARG);
+    PIceCandidate pIceCandidate = pIceAgent->pDataSendingIceCandidatePair->remote;
+    PRtcIceCandidateDiagnostics pRtcIceCandidateDiagnostics = &pIceAgent->rtcSelectedRemoteIceCandidateDiagnostics;
+    if (!isRemote) {
+        pIceCandidate = pIceAgent->pDataSendingIceCandidatePair->local;
+        pRtcIceCandidateDiagnostics = &pIceAgent->rtcSelectedLocalIceCandidateDiagnostics;
+        // URL and relay protocol are populated only for local candidate by spec
+        STRNCPY(pRtcIceCandidateDiagnostics->url, pIceAgent->iceServers[pIceCandidate->iceServerIndex].url,
```

Comment:
what happens for host candidates?

Reply:
I updated to set the url to N/A for host candidates. If it is index 0, it would choose STUN server URL.